From d61e6a19c967943ffe1dfe68662e63f0fd2a3949 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Fri, 14 Nov 2025 17:12:09 +0100
Subject: [PATCH] cloud-ssh-keys-fetcher: allow timeout to create a new process
 group

Allow the timeout utility to create a new process group by calling
setpgid. This is needed so timeout can collect all child processes that
may have been spawned when the timeout expires.

Currently, timeout is only used by EC2 Instance Connect which is invoked
by cloud-ssh-keys-fetcher. If and when EC2 Instance Connect gets its own
domain, this permission can move.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/modules/contrib/cloud-ssh-keys-fetcher.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/contrib/cloud-ssh-keys-fetcher.te b/policy/modules/contrib/cloud-ssh-keys-fetcher.te
index 73cc06db6..c9cc9268a 100644
--- a/policy/modules/contrib/cloud-ssh-keys-fetcher.te
+++ b/policy/modules/contrib/cloud-ssh-keys-fetcher.te
@@ -41,3 +41,6 @@ miscfiles_read_generic_certs(cloud_ssh_keys_fetcher_t)
 # curl: call cloud-specific meta-data services
 allow cloud_ssh_keys_fetcher_t self:tcp_socket create_socket_perms;
 corenet_tcp_connect_http_port(cloud_ssh_keys_fetcher_t)
+
+# timeout: new process group to collect all child processes
+allow cloud_ssh_keys_fetcher_t self:process { setpgid };
-- 
2.51.2

